### ISSUE TITLE
Adding object to already existing nwb file

### ISSUE THREAD

**Issue by GoktugAlkan** [NeurodataWithoutBorders/matnwb#502]
Hello,

I am trying to figure out if there is a function that can be used to read an nwb file and add new information to this and save this modified nwb file at the same path again.

I created a large nwb file with an extracellular recording. We extracted the waveforms of the spiking cells and want to add information about these units by using `nwb.units`. It would be very convenient if we would not need to write the whole data from scratch again. Instead, we would like to append `nwb.units` to our already existing nwb file.

I tried to check with the previous issues mentioning this, but it was not very clear to me if there is a function that can be used as in `pynwb`.

Many thanks in advance!

**Comment by GoktugAlkan** [comment]
Thanks for the response @lawrence-mbf. The already existing nwb file doesn't contain anything in `nwb.units`, i.e., this field is empty. Nothing has been initialized in that field. 

Is there a way to populate that field? I.e., is there a way to read the already existing nwb file and to add data to the field `nwb.units`?

**Comment by GoktugAlkan** [comment]
```
%% read pre-existing nwb file
filepathRead = '/home/matlab/Docker_shared/NWB Project/testAppendingData/testAppend.nwb';
readFile = nwbRead(filepath);

%% create dummy time series
dummyData = types.core.TimeSeries( ...
    'starting_time', 0.0, ... % seconds
    'starting_time_rate', 30000., ... % Hz
    'data', randn(12, 3000));

%% populate readfile.acquisition with dummy time series
readFile.acquisition.set('dummyTimeSeries', dummyData);

%% store 
filepathStore = '/home/matlab/Docker_shared/NWB Project/testAppendingData/testAppend.nwb';
nwbExport(readFile, filepath)
```

**Comment by GoktugAlkan** [comment]
@lawrence-mbf I tried to implement your proposed approach. Below is the code that I'll try to explain also more in detail.

```
%% read pre-existing nwb file
filepathRead = '/home/matlab/Docker_shared/NWB Project/testAppendingData/testAppend.nwb';
readFile = nwbRead(filepath);

%% create dummy time series
dummyData = types.core.TimeSeries( ...
    'starting_time', 0.0, ... % seconds
    'starting_time_rate', 30000., ... % Hz
    'data', randn(12, 3000));

%% populate readfile.acquisition with dummy time series
readFile.acquisition.set('dummyTimeSeries', dummyData);

%% store modified nwb file
filepathStoreA = '/home/matlab/Docker_shared/NWB Project/testAppendingData/testAppend.nwb';
% filepathStoreB = '/home/matlab/Docker_shared/NWB Project/testAppendingData/testAppend_B.nwb';
nwbExport(readFile, filepath)
```
Below is an explanation of what I am doing in thhe code from above:

1.  In the first section (indicated by `%% read pre-existing nwb file`, the pre-existing nwb file is imported. It has just one populated field, namely `nwb.processing`.
2. In the second section (indicated by `%% create dummy time series`), a dummy time series is created.
3. In the third section (indicated by `%% populate readfile.acquisition with dummy time series`), the acquistion field is populated with the dummy time series.
4. In the last section (indicate by `%% store modified nwb file`), the modified nwb file is stored. With filepathStoreA, I am trying to store the modified nwb file at the same path as the pre-exisiting nwb file that was imported in step 1. However, this is not working. Also using filepathStoreB, where I try to store the modified nwb file, at another path is also not working. 

The two cases described in step 4 lead me to the same error message:

```
Error using hdf5lib2
The HDF5 library encountered an unknown error.

Error in H5A.write (line 53)
H5ML.hdf5lib2('H5Awrite', attr_id, type_id, buf);

Error in io.writeAttribute (line 25)
H5A.write(id, tid, data);

Error in types.core.TimeSeries/export (line 381)
            io.writeAttribute(fid, [fullpath '/data/unit'], obj.data_unit);

Error in types.untyped.Set/export (line 181)
                    refs = v.export(fid, propfp, refs);

Error in types.core.NWBFile/export (line 834)
        refs = obj.acquisition.export(fid, [fullpath '/acquisition'], refs);

Error in NwbFile/export (line 62)
                refs = export@types.core.NWBFile(obj, output_file_id, '/', {});

Error in nwbExport (line 36)
    nwb(i).export(filename);

Error in untitled15 (line 17)
nwbExport(readFile, filepathStoreA)
```

Do you have an idea what goes wrong here?

Thanks in advance!

**Comment by lawrence-mbf** [comment]
@GoktugAlkan ,

I am unable to create this issue at the tip of master. Are you sure there's no other data in the `/acquisition` folder?
Also, just to make sure, what version of MatNWB are you using?

**Comment by lawrence-mbf** [comment]
This looks like the issue was resolved in #503 . If not, feel free to reopen.
